From 7e003a418bb69c2c331a1c589c01afc5ad06b6a5 Mon Sep 17 00:00:00 2001
From: Elena Taivan <elena.taivan@windriver.com>
Date: Wed, 30 Sep 2020 14:14:32 +0000
Subject: [PATCH] Wrong usage of 'rbd_store_chunk_size'

'rbd_store_chunk_size' option represents the size of the chunks
of the objects into which an image is chuncked.
It does not represent the 'pg_num' value of 'images' ceph pool.

Solution: replace 'rdb_store_chunk_size' with 'chunk_size' custom
option.

[ updated glance_store chunk_size for epoxy release ]
Signed-off-by: cfigueir <carlos.figueiredofelipe@windriver.com>
---
 glance/templates/job-storage-init.yaml | 2 +-
 glance/values.yaml                     | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/glance/templates/job-storage-init.yaml b/glance/templates/job-storage-init.yaml
index be9d2331b..d24a0cfb3 100644
--- a/glance/templates/job-storage-init.yaml
+++ b/glance/templates/job-storage-init.yaml
@@ -122,7 +122,7 @@ spec:
             - name: RBD_POOL_CRUSH_RULE
               value: {{ .Values.conf.glance.rbd.rbd_store_crush_rule | quote }}
             - name: RBD_POOL_CHUNK_SIZE
-              value: {{ .Values.conf.glance.rbd.rbd_store_chunk_size | quote }}
+              value: {{ .Values.conf.glance.glance_store.chunk_size | quote }}
             - name: RBD_POOL_SECRET
               value: {{ .Values.secrets.rbd | quote }}
             {{ end }}
diff --git a/glance/values.yaml b/glance/values.yaml
index 8e1d1371b..d545e24b3 100644
--- a/glance/values.yaml
+++ b/glance/values.yaml
@@ -265,7 +265,7 @@ conf:
       # This is for backward compatibility.
       filesystem_store_datadir: /var/lib/glance/images
       cinder_catalog_info: volumev3::internalURL
-      rbd_store_chunk_size: 8
+      chunk_size: 8
       rbd_store_replication: 3
       rbd_store_crush_rule: replicated_rule
       rbd_store_pool: glance.images
-- 
2.43.0

